### PR TITLE
feature(templateDesigner): #17 Presenting / Interactive grid layer ha…

### PR DESCRIPTION
**Pull request by derpdead**
…d been implemented.

### CODE REVIEW

File: components/Grid/Template/TemplateGrid.vue [hunk 1]
```diff
@@ -80,14 +97,29 @@ export default {
...
         },
+        draggableGridAreaLenght() {
+            return this.maxRows * this.columnsNumber;
+        },
         errorMessages() {
             return this.titleValidationError ? [this.titleValidationError] : null;
         },
+        gridStyles() {
+            return {
+                gridTemplateColumns: 'repeat(4, 1fr)',
```
Comment (bleto) [inline components/Grid/Template/TemplateGrid.vue]: I think it should be related with **columnsNumber** variable.

File: components/TemplateGrid/TemplateDesigner/TemplateGridGhostItem.vue [hunk 2]
```diff
@@ -0,0 +1,56 @@
+<template>
```
Comment (bleto) [inline components/TemplateGrid/TemplateDesigner/TemplateGridGhostItem.vue]: You forgot about license header.

File: components/Grid/Template/TemplateGrid.vue [hunk 3]
```diff
@@ -6,17 +6,31 @@
...
+                    :rows="maxRows"
+                    :is-header="false" />
+                <div
+                    :style="gridStyles"
+                    class="draggable-grid">
+                    <template v-for="item in draggableGridAreaLenght">
+                        <TemplateGridGhostItem
+                            :key="item"
+                            :position="{
+                                row: Math.ceil(item / columnsNumber),
```
Comment (tomaszstrozik) [inline components/Grid/Template/TemplateGrid.vue]: consider to put 'row' and 'column' expression to function with proper name in component

File: components/Grid/Template/TemplateGrid.vue [hunk 4]
```diff
@@ -31,6 +45,9 @@ import AttributeElement from '~/components/Template/AttributeElement';
 export default {
     name: 'TemplateGrid',
     components: {
+        TemplateGridPresentationLayer: () => import('~/components/TemplateGrid/TemplateGridPresentationLayer'),
```
Comment (tomaszstrozik) [inline components/Grid/Template/TemplateGrid.vue]: is it necessary to dynamically import those components?

File: components/Grid/Template/TemplateGrid.vue [hunk 5]
```diff
@@ -80,14 +100,29 @@ export default {
...
+        },
     },
     methods: {
         ...mapActions('templateDesigner', [
             'setTemplateDesignerTitle',
         ]),
+        onDrop(position) {
+            this.insertedItems.push({ ...position });
+        },
+        onRowsCountChange({ /** key, */ value }) {
```
[the remark this answers is not in the document]
Comment (derpdead) [reply components/Grid/Template/TemplateGrid.vue]: It's going to be removed after all feature is going to be implemented - reminded

File: components/Grid/Template/TemplateGrid.vue (hunk 3 above)
Comment (derpdead) [reply components/Grid/Template/TemplateGrid.vue]: I did it in next PR 💃

**Pull request merged**